projects/plugins/jetpack/tests/e2e/lib/env/prerequisites.js

```diff
@@ -0,0 +1,170 @@
+import logger from '../logger';
+import { isBlogTokenSet, syncJetpackPlanData } from '../flows/jetpack-connect';
+import {
+	execMultipleWpCommands,
+	execWpCommand,
+	getAccountCredentials,
+	provisionJetpackStartConnection,
+	resetWordpressInstall,
+} from '../utils-helper';
+import fs from 'fs';
+import config from 'config';
+import { loginToWpCom, loginToWpSite } from '../flows/log-in';
+
+export function prerequisitesBuilder() {
+	const state = {
+		loggedIn: undefined,
+		wpComLoggedIn: undefined,
+		connected: undefined,
+		plan: undefined,
+		modules: { active: undefined, inactive: undefined },
+		clean: undefined,
+	};
+
+	return {
+		withLoggedIn( shouldBeLoggedIn ) {
```

Comment:
What was your motivation to prefix builder methods with `with`? 

I wonder if it might be clearer without it:

```prerequisitesBuilder().
.SiteLogin(false)
.WpComLogin(true)
.UserConnection(true)
.ActivateModules([module_a, module_b])
.build();
```

Not a big deal though, so feel free to ignore it :)